digraph "unitGraph" {
    "r14 := @parameter0: java.lang.String[]"
    "$r0 = new java.util.concurrent.ForkJoinPool"
    "specialinvoke $r0.<java.util.concurrent.ForkJoinPool: void <init>()>()"
    "r1 = $r0"
    "$r2 = new ch3.s2.CountTask"
    "specialinvoke $r2.<ch3.s2.CountTask: void <init>(long,long)>(0L, 200000L)"
    "r3 = $r2"
    "r4 = virtualinvoke r1.<java.util.concurrent.ForkJoinPool: java.util.concurrent.ForkJoinTask submit(java.util.concurrent.ForkJoinTask)>(r3)"
    "$r5 = virtualinvoke r4.<java.util.concurrent.ForkJoinTask: java.lang.Object get()>()"
    "$r6 = (java.lang.Long) $r5"
    "l0 = virtualinvoke $r6.<java.lang.Long: long longValue()>()"
    "$r8 = <java.lang.System: java.io.PrintStream out>"
    "$r7 = new java.lang.StringBuilder"
    "specialinvoke $r7.<java.lang.StringBuilder: void <init>()>()"
    "$r9 = virtualinvoke $r7.<java.lang.StringBuilder: java.lang.StringBuilder append(java.lang.String)>(\"sum=\")"
    "$r10 = virtualinvoke $r9.<java.lang.StringBuilder: java.lang.StringBuilder append(long)>(l0)"
    "$r11 = virtualinvoke $r10.<java.lang.StringBuilder: java.lang.String toString()>()"
    "virtualinvoke $r8.<java.io.PrintStream: void println(java.lang.String)>($r11)"
    "goto [?= return]"
    "$r13 := @caughtexception"
    "r15 = $r13"
    "virtualinvoke r15.<java.lang.InterruptedException: void printStackTrace()>()"
    "$r12 := @caughtexception"
    "r16 = $r12"
    "virtualinvoke r16.<java.util.concurrent.ExecutionException: void printStackTrace()>()"
    "return"
    "r14 := @parameter0: java.lang.String[]"->"$r0 = new java.util.concurrent.ForkJoinPool";
    "$r0 = new java.util.concurrent.ForkJoinPool"->"specialinvoke $r0.<java.util.concurrent.ForkJoinPool: void <init>()>()";
    "specialinvoke $r0.<java.util.concurrent.ForkJoinPool: void <init>()>()"->"r1 = $r0";
    "r1 = $r0"->"$r2 = new ch3.s2.CountTask";
    "$r2 = new ch3.s2.CountTask"->"specialinvoke $r2.<ch3.s2.CountTask: void <init>(long,long)>(0L, 200000L)";
    "specialinvoke $r2.<ch3.s2.CountTask: void <init>(long,long)>(0L, 200000L)"->"r3 = $r2";
    "r3 = $r2"->"r4 = virtualinvoke r1.<java.util.concurrent.ForkJoinPool: java.util.concurrent.ForkJoinTask submit(java.util.concurrent.ForkJoinTask)>(r3)";
    "r4 = virtualinvoke r1.<java.util.concurrent.ForkJoinPool: java.util.concurrent.ForkJoinTask submit(java.util.concurrent.ForkJoinTask)>(r3)"->"$r13 := @caughtexception";
    "r4 = virtualinvoke r1.<java.util.concurrent.ForkJoinPool: java.util.concurrent.ForkJoinTask submit(java.util.concurrent.ForkJoinTask)>(r3)"->"$r12 := @caughtexception";
    "r4 = virtualinvoke r1.<java.util.concurrent.ForkJoinPool: java.util.concurrent.ForkJoinTask submit(java.util.concurrent.ForkJoinTask)>(r3)"->"$r5 = virtualinvoke r4.<java.util.concurrent.ForkJoinTask: java.lang.Object get()>()";
    "$r5 = virtualinvoke r4.<java.util.concurrent.ForkJoinTask: java.lang.Object get()>()"->"$r13 := @caughtexception";
    "$r5 = virtualinvoke r4.<java.util.concurrent.ForkJoinTask: java.lang.Object get()>()"->"$r12 := @caughtexception";
    "$r5 = virtualinvoke r4.<java.util.concurrent.ForkJoinTask: java.lang.Object get()>()"->"$r6 = (java.lang.Long) $r5";
    "$r6 = (java.lang.Long) $r5"->"$r13 := @caughtexception";
    "$r6 = (java.lang.Long) $r5"->"$r12 := @caughtexception";
    "$r6 = (java.lang.Long) $r5"->"l0 = virtualinvoke $r6.<java.lang.Long: long longValue()>()";
    "l0 = virtualinvoke $r6.<java.lang.Long: long longValue()>()"->"$r13 := @caughtexception";
    "l0 = virtualinvoke $r6.<java.lang.Long: long longValue()>()"->"$r12 := @caughtexception";
    "l0 = virtualinvoke $r6.<java.lang.Long: long longValue()>()"->"$r8 = <java.lang.System: java.io.PrintStream out>";
    "$r8 = <java.lang.System: java.io.PrintStream out>"->"$r13 := @caughtexception";
    "$r8 = <java.lang.System: java.io.PrintStream out>"->"$r12 := @caughtexception";
    "$r8 = <java.lang.System: java.io.PrintStream out>"->"$r7 = new java.lang.StringBuilder";
    "$r7 = new java.lang.StringBuilder"->"$r13 := @caughtexception";
    "$r7 = new java.lang.StringBuilder"->"$r12 := @caughtexception";
    "$r7 = new java.lang.StringBuilder"->"specialinvoke $r7.<java.lang.StringBuilder: void <init>()>()";
    "specialinvoke $r7.<java.lang.StringBuilder: void <init>()>()"->"$r13 := @caughtexception";
    "specialinvoke $r7.<java.lang.StringBuilder: void <init>()>()"->"$r12 := @caughtexception";
    "specialinvoke $r7.<java.lang.StringBuilder: void <init>()>()"->"$r9 = virtualinvoke $r7.<java.lang.StringBuilder: java.lang.StringBuilder append(java.lang.String)>(\"sum=\")";
    "$r9 = virtualinvoke $r7.<java.lang.StringBuilder: java.lang.StringBuilder append(java.lang.String)>(\"sum=\")"->"$r13 := @caughtexception";
    "$r9 = virtualinvoke $r7.<java.lang.StringBuilder: java.lang.StringBuilder append(java.lang.String)>(\"sum=\")"->"$r12 := @caughtexception";
    "$r9 = virtualinvoke $r7.<java.lang.StringBuilder: java.lang.StringBuilder append(java.lang.String)>(\"sum=\")"->"$r10 = virtualinvoke $r9.<java.lang.StringBuilder: java.lang.StringBuilder append(long)>(l0)";
    "$r10 = virtualinvoke $r9.<java.lang.StringBuilder: java.lang.StringBuilder append(long)>(l0)"->"$r13 := @caughtexception";
    "$r10 = virtualinvoke $r9.<java.lang.StringBuilder: java.lang.StringBuilder append(long)>(l0)"->"$r12 := @caughtexception";
    "$r10 = virtualinvoke $r9.<java.lang.StringBuilder: java.lang.StringBuilder append(long)>(l0)"->"$r11 = virtualinvoke $r10.<java.lang.StringBuilder: java.lang.String toString()>()";
    "$r11 = virtualinvoke $r10.<java.lang.StringBuilder: java.lang.String toString()>()"->"$r13 := @caughtexception";
    "$r11 = virtualinvoke $r10.<java.lang.StringBuilder: java.lang.String toString()>()"->"$r12 := @caughtexception";
    "$r11 = virtualinvoke $r10.<java.lang.StringBuilder: java.lang.String toString()>()"->"virtualinvoke $r8.<java.io.PrintStream: void println(java.lang.String)>($r11)";
    "virtualinvoke $r8.<java.io.PrintStream: void println(java.lang.String)>($r11)"->"$r13 := @caughtexception";
    "virtualinvoke $r8.<java.io.PrintStream: void println(java.lang.String)>($r11)"->"$r12 := @caughtexception";
    "virtualinvoke $r8.<java.io.PrintStream: void println(java.lang.String)>($r11)"->"goto [?= return]";
    "goto [?= return]"->"return";
    "$r13 := @caughtexception"->"r15 = $r13";
    "r15 = $r13"->"virtualinvoke r15.<java.lang.InterruptedException: void printStackTrace()>()";
    "virtualinvoke r15.<java.lang.InterruptedException: void printStackTrace()>()"->"goto [?= return]";
    "goto [?= return]"->"return";
    "$r12 := @caughtexception"->"r16 = $r12";
    "r16 = $r12"->"virtualinvoke r16.<java.util.concurrent.ExecutionException: void printStackTrace()>()";
    "virtualinvoke r16.<java.util.concurrent.ExecutionException: void printStackTrace()>()"->"return";
}
